src/MongoDB.EntityFrameworkCore/Infrastructure/MongoOptionsExtension.cs

```diff
@@ -96,45 +121,88 @@ public virtual MongoOptionsExtension WithDatabaseName(string databaseName)
         return clone;
     }
 
+    /// <summary>
+    /// Obtains the current key value namespace if one is specified, otherwise null.
+    /// </summary>
+    public CollectionNamespace? KeyVaultNamespace { get; private set; }
 
     /// <summary>
-    /// Obtains the current <see cref="IMongoClient"/> if one is specified, otherwise null.
+    /// Specifies a key vault namespace to identify keys on the MongoDB server.
     /// </summary>
-    public IMongoClient? MongoClient { get; private set; }
+    /// <param name="keyVaultNamespace">The name of the database to use.</param>
+    /// <returns>The <see cref="MongoOptionsExtension"/> to continue chaining configuration.</returns>
+    public virtual MongoOptionsExtension WithKeyVaultNamespace(CollectionNamespace? keyVaultNamespace)
+    {
+        var clone = Clone();
+        clone.KeyVaultNamespace = keyVaultNamespace;
+        return clone;
+    }
 
     /// <summary>
-    /// Specify a <see cref="IMongoClient"/> to use for connecting to the MongoDB server.
+    /// Obtains the current KMS Providers to use for data encryption if specified, otherwise null.
     /// </summary>
-    /// <param name="mongoClient">The <see cref="IMongoClient"/> to use when communicating with the MongoDB server.</param>
+    public Dictionary<string, IReadOnlyDictionary<string, object>>? KmsProviders { get; private set; }
+
+    /// <summary>
+    /// Specifies KMS Providers to use for data encryption.
+    /// </summary>
+    /// <param name="kmsProviders">The KMS Providers to use.</param>
     /// <returns>The <see cref="MongoOptionsExtension"/> to continue chaining configuration.</returns>
-    public virtual MongoOptionsExtension WithMongoClient(IMongoClient mongoClient)
+    public virtual MongoOptionsExtension WithKmsProviders(Dictionary<string, IReadOnlyDictionary<string, object>>? kmsProviders)
     {
-        ArgumentNullException.ThrowIfNull(mongoClient);
-        EnsureConnectionNotAlreadyConfigured(nameof(MongoClient));
+        var clone = Clone();
+        clone.KmsProviders = kmsProviders;
+        return clone;
+    }
+
+    /// <summary>
+    /// Obtains the current encryption provider.
```

Comment:
"Gets" is normal. "Returns" is normal. "Obtains" is...weird. :-) I mean, it's fine, just unusual.